### ISSUE TITLE
Incorrect loss calculation after v1.10: fitting tf.Dataset in Keras.

### ISSUE THREAD

**Issue by skiler07** [tensorflow/tensorflow#23634]
In 1.12, we can pass `tf.Dataset` into our Keras models. For instance, I'm modelling a binary classification with high class imbalance. 

```
def _parse_func(record):
    keys_to_features = {} # some schema

    parsed = tf.parse_single_example(record, keys_to_features)

    # Build the X output
    x = parsed['x']  
    y = parsed['y']
    w = parsed['w'] * DEBUG_WEIGHT

    return x, y, w
```
where `DEBUG_WEIGHT` is changed from 1 to 10000, with no affect on the loss.
Create `tf.Dataset`

```
def input_fn(f, b=BATCH_SIZE):
    """
    :param f: A list of file names e.g. ['data-0.tf-record', 'data-1.tf-record'] to read
    :param b: Batch size, defaults to BATCH_SIZE in hparams.py
    :return: And infinitely iterable data set using tf records of tf.data.Dataset class
    """
    d = tf.data.TFRecordDataset(filenames=f)
    d = d.map(map_func=_parse_func)
    d = d.repeat()
    d = d.batch(batch_size=b)
    return d
```

But when I fit the model using Keras, no matter what weight is passed, it seems ignored.

E.g.

```
    train_data = input_fn(f=['path_to_tf_records'])

    params = {
        'loss': 'binary_crossentropy',
        'optimizer': optimizer,
        'metrics': metrics,
    }
    model.compile(**params)

    model.fit(x=train_data.make_one_shot_iterator(),
              epochs=50,
              steps_per_epoch=10000)
```
Interestingly, if I pass `class_weight` to the `fit`, then I get the message that both `sample_weights` and `class_weight` are provided. So why doesn't the loss change?

The loss is incredibly small and it doesn't change with weight.

Epoch 1/30
2018-11-09 16:24:10.605651: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
 461/9375 [>.............................] - ETA: 2:01:50 - loss: 53.4499 - binary_accuracy: 0.8183 - binary_crossentropy: 0.4290 - mean_absolute_error: 0.3171

## Sidenote
When I used numpy arrays with `tf.keras.utils.Sequence` class as a generator, using `fit_generator`, the loss was also not changing properly. 

### Takeaways

Clearly, the weight is not passed to the loss function correctly. Even if I use `fit_generator()` the loss differs in tensor flow > 1.10. Additionally, workers must be set to 0 i.e. `workers=0` for the generator to work which is incredibly slow during training.
